htdocs/admin/fediverse.php

```diff
@@ -252,6 +273,53 @@
 print '<td>https://mastodon.social/api/v1/accounts/id_user</td>';
 print '</tr>';
 
+print '<script>
+$(document).ready(function() {
+    function toggleOAuthServiceDisplay() {
+        if ($("#radio_oauth").is(":checked")) {
+            $("#oauth_service_div").show();
+        } else {
+            $("#oauth_service_div").hide();
+        }
+    }
+
+    toggleOAuthServiceDisplay();
+
+    $("input[name=\"'.$vartosmtpstype.'\"]").change(function() {
+        toggleOAuthServiceDisplay();
+    });
+});
+</script>';
+// Methodes oauth
+print '<td>'.$langs->trans("MAIN_MAIL_SMTPS_AUTH_TYPE").'</td>';
+print '<td>';
+print '<input type="radio" id="radio_oauth" name="'.$vartosmtpstype.'" value="XOAUTH2"'.(getDolGlobalString($vartosmtpstype) == 'XOAUTH2' ? ' checked' : '').(isModEnabled('oauth') ? '' : ' disabled').'>';
+print '<label for="radio_oauth">'.$form->textwithpicto($langs->trans("UseOauth"), $langs->trans("OauthNotAvailableForAllAndHadToBeCreatedBefore")).'</label>';
+if (!isModEnabled('oauth')) {
+	print ' &nbsp; <a href="'.DOL_URL_ROOT.'/admin/modules.php?search_keyword=oauth">'.$langs->trans("EnableModuleX", "OAuth").'</a>';
+} else {
+	print ' &nbsp; <a href="'.DOL_URL_ROOT.'/admin/oauth.php">'.$langs->trans("SetupModuleX", " OAuth").'</a>';
+}
+print '</td>';
+print '</tr>';
+
+print '<tr class="oddeven" id="oauth_service_div"  style="display: none;">';
+print '<td>'.$langs->trans("OAuthService").'</td>';
+print '<td>';
+if (!isModEnabled('multicompany') || ($user->admin && !$user->entity)) {
+	print $form->selectarray('OAUTH_SERVICE_SOCIAL_NETWORK', $oauthservices, $conf->global->OAUTH_SERVICE_SOCIAL_NETWORK);
```

Comment:
I suppose that getDolGlobalString() will help for phan, and also typing $oauthservices for phan.
```
'@phan-var-force array{field1:type,field2:type,...} $oauthservices';
```
 just after the loop where it is dynamically defined.

Reply:
ok i see, thanks

Reply:
@lamrani002 why you check if multicompany is not enabled ? Wht is the problem ?

`if (!isModEnabled('multicompany') || ($user->admin && !$user->entity)) {`

Reply:
@lamrani002 As far as I can tell, after adding the typing hint, the phan notice is now there because one of the items in the array is an int and 'selectarray' is said to expect string[] (when it's an array).  So either convert all items to string, or update the selectarray() param type to 'int|string|array<int|string>'